frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -142,141 +173,218 @@ function ProblemDisplay(props: ProblemDisplayParams) {
+                  e.target.value, newProblem.problemInputs[index].type)}
+              />
 
-                <label htmlFor={`problem-hidden-${index}`}>
-                  Hidden
-                  <CheckboxInput
-                    id={`problem-hidden-${index}`}
-                    checked={newProblem.testCases[index].hidden}
-                    onChange={(e) => {
-                      const current = newProblem.testCases[index];
-                      handleTestCaseChange(index, current.input,
-                        current.output, e.target.checked, current.explanation);
-                    }}
-                  />
-                </label>
+              <PrimarySelect
+                onChange={(e) => handleInputChange(
+                  index,
+                  newProblem.problemInputs[index].name,
+                  ProblemIOType[e.target.value as keyof typeof ProblemIOType],
+                )}
+                value={problemIOTypeToString(newProblem.problemInputs[index].type)}
+              >
+                {
+                  Object.keys(ProblemIOType).map((key) => (
+                    <option key={key} value={key}>{key}</option>
+                  ))
+                }
+              </PrimarySelect>
 
-                <SmallButton onClick={() => deleteTestCase(index)}>Delete Test Case</SmallButton>
-              </div>
-            ))}
-            <SmallButton onClick={addTestCase}>Add Test Case</SmallButton>
-          </div>
-        ) : null}
+              <TextButton
+                style={{
+                  display: 'inline-block',
+                  marginLeft: '2.5px',
+                }}
+                color={ThemeConfig.colors.gray}
+                onClick={() => deleteProblemInput(index)}
+              >
+                ✕
+              </TextButton>
+            </div>
+          ))}
+          <TextButton
+            color={ThemeConfig.colors.gray}
+            onClick={addProblemInput}
+          >
+            Add +
+          </TextButton>
 
-      <LargeInputButton value={actionText} onClick={() => onClick(newProblem)} />
-      {editMode ? <LargeInputButton value="Delete Problem" onClick={deleteProblemFunc} /> : null}
+          <LowMarginMediumText>Problem Output</LowMarginMediumText>
+          <PrimarySelect
+            onChange={(e) => handleEnumChange('outputType', ProblemIOType[e.target.value as keyof typeof ProblemIOType])}
+            value={problemIOTypeToString(newProblem.outputType)}
+          >
+            {
+              Object.keys(ProblemIOType).map((key) => (
+                <option key={key} value={key}>{key}</option>
+              ))
+            }
+          </PrimarySelect>
 
-      {loading ? <Loading /> : null}
-      {error ? <ErrorMessage message={error} /> : null}
-    </Content>
+          {editMode
+            ? (
+              <>
+                <LowMarginMediumText>Danger Zone</LowMarginMediumText>
```

Comment:
I like that danger zone haha

Reply:
Yeah haha, it's on GitHub and I think other sites have adopted it too (like Gimkit actually lol) 😁